crates/uv-configuration/src/dev.rs

```diff
@@ -121,70 +131,120 @@ impl DevGroupsSpecification {
-            groups,
-        }
+        Self { dev, groups }
     }
 
     /// Return a new [`DevGroupsSpecification`] with development dependencies included by default.
     ///
     /// This is appropriate in projects, where the `dev` group is synced by default.
     #[must_use]
-    pub fn with_default_dev(self) -> Self {
-        match self.dev {
-            Some(_) => self,
-            None => match self.groups {
-                // Only include the default `dev` group if `--only-group` wasn't used
-                GroupsSpecification::Only(_) => self,
-                GroupsSpecification::Exclude | GroupsSpecification::Include(_) => Self {
-                    dev: Some(DevMode::Include),
-                    ..self
-                },
-            },
+    pub fn with_defaults(self, defaults: Vec<GroupName>) -> DevGroupsManifest {
+        DevGroupsManifest {
+            spec: self,
+            defaults,
         }
     }
 
     /// Returns `true` if the specification allows for production dependencies.
     pub fn prod(&self) -> bool {
-        (self.dev.is_none() || self.dev.as_ref().is_some_and(DevMode::prod)) && self.groups.prod()
+        self.dev.as_ref().map_or(true, DevMode::prod)
+            && self.groups.as_ref().map_or(true, GroupsSpecification::prod)
     }
 
-    /// Returns the flag that was used to request development dependencies.
+    /// Returns `true` if the specification is limited to a select set of groups.
+    pub fn only(&self) -> bool {
+        self.dev.as_ref().is_some_and(DevMode::only)
+            || self.groups.as_ref().is_some_and(GroupsSpecification::only)
+    }
+
+    /// Returns the flag that was used to request development dependencies, if specified.
     pub fn dev_mode(&self) -> Option<&DevMode> {
         self.dev.as_ref()
     }
 
-    /// Returns the list of groups to include.
-    pub fn groups(&self) -> &GroupsSpecification {
-        &self.groups
+    /// Returns the list of groups to include, if specified.
+    pub fn groups(&self) -> Option<&GroupsSpecification> {
+        self.groups.as_ref()
     }
 }
 
 impl From<DevMode> for DevGroupsSpecification {
     fn from(dev: DevMode) -> Self {
         Self {
             dev: Some(dev),
-            groups: GroupsSpecification::Exclude,
+            groups: None,
         }
     }
 }
 
 impl From<GroupsSpecification> for DevGroupsSpecification {
     fn from(groups: GroupsSpecification) -> Self {
-        Self { dev: None, groups }
+        Self {
+            dev: None,
+            groups: Some(groups),
+        }
+    }
+}
+
+/// The manifest of `dependency-groups` to include, taking into account the user-provided
+/// [`DevGroupsSpecification`] and the project-specific default groups.
+#[derive(Debug, Clone)]
+pub struct DevGroupsManifest {
+    /// The specification for the development dependencies.
+    pub(crate) spec: DevGroupsSpecification,
+    /// The default groups to include.
+    pub(crate) defaults: Vec<GroupName>,
+}
```

Comment:
I don't feel strongly on the name here, but this is the specification along with the project-specific defaults.